src/Plugins/Nop.Plugin.ExternalAuth.Facebook/Controllers/FacebookDataDeletionController.cs

```diff
@@ -0,0 +1,153 @@
+        private readonly ILogger _logger;
+        private readonly INotificationService _notificationService;
+        private readonly IUrlHelperFactory _urlHelperFactory;
+        private readonly IWebHelper _webHelper;
+
+        #endregion
+
+        #region Ctor
+
+        public FacebookDataDeletionController(FacebookExternalAuthSettings facebookExternalAuthSettings,
+            IActionContextAccessor actionContextAccessor,
+            IExternalAuthenticationService externalAuthenticationService,
+            ILocalizationService localizationService,
+            ILogger logger,
+            INotificationService notificationService,
+            IUrlHelperFactory urlHelperFactory,
+            IWebHelper webHelper)
+        {
+            _facebookExternalAuthSettings = facebookExternalAuthSettings;
+            _actionContextAccessor = actionContextAccessor;
+            _externalAuthenticationService = externalAuthenticationService;
+            _localizationService = localizationService;
+            _logger = logger;
+            _notificationService = notificationService;
+            _urlHelperFactory = urlHelperFactory;
+            _webHelper = webHelper;
+        }
+
+        #endregion
+
+        #region Utilities
+
+        // Convert string to a valid Base64 encoded string
+        protected static string DecodeUrlBase64(string str)
+        {
+            if(string.IsNullOrEmpty(str))
+                return null;
+
+            str = str.Replace("-", "+").Replace("_", "/");
+            var paddingToAdd = (str.Length % 4) == 3 ? 1 : (str.Length % 4);
+            var charToAdd = new string('=', paddingToAdd);
+
+            return str += charToAdd;
+        }
+
+        #endregion
+
+        #region Methods
+        
+        [HttpPost]
+        public async Task<IActionResult> DataDeletionCallback(IFormCollection form)
+        {
+            try
+            {
+                string signed_request = form["signed_request"];
+
+                if (!string.IsNullOrEmpty(signed_request))
+                {
+                    var split = signed_request.Split('.');
+
+                    var signatureRaw = DecodeUrlBase64(split[0]);
+
+                    if (string.IsNullOrEmpty(signatureRaw))
+                        return null;
+
+                    var dataRaw = DecodeUrlBase64(split[1]);
+
+                    if (string.IsNullOrEmpty(dataRaw))
+                        return null;
+
+                    // the decoded signature
+                    var signature = Convert.FromBase64String(signatureRaw);
+                    var dataBuffer = Convert.FromBase64String(dataRaw);
+
+                    // JSON object
+                    var json = Encoding.UTF8.GetString(dataBuffer);
+
+                    var appSecretBytes = Encoding.UTF8.GetBytes(_facebookExternalAuthSettings.ClientSecret);
+                    HMAC hmac = new HMACSHA256(appSecretBytes);
```

Comment:
The algorithm will always be HMACSHA256 or is it passed in the request (`algorithm` field)?

Reply:
// confirm the signature
  $expected_sig = hash_hmac('sha256', $payload, $secret, $raw = true);

this code snippet is from Facebook documentation. So we can say it will always be HMACSHA256